### PR TITLE
Add SyncThriftClientErrorHandler for proxy to clean the code

### PR DESCRIPTION
**Pull request by zjx1230**
Add SyncThriftClientErrorHandler for proxy to clean the code.

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/client/ClientManager.java [hunk 1]
```diff
@@ -56,7 +56,9 @@ public V borrowClient(K node) throws IOException {
       throw e;
     } catch (Exception e) {
       // external doesn't care of other exceptions
-      logger.error(String.format("Borrow client from pool for node %s failed.", node), e);
+      String errorMessage = String.format("Borrow client from pool for node %s failed.", node);
+      logger.error(errorMessage, e);
+      throw new IOException(errorMessage, e);
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/client/ClientManager.java:61]: Can you check all existing places where clients are obtained through clientManager and remove null judgments? As far as I know, ConfigNode is also using clientManager

Comment (zjx1230) [reply node-commons/src/main/java/org/apache/iotdb/commons/client/ClientManager.java:61]: Do you want me to try to remove all the relevant null judgment?

Comment (OneSizeFitsQuorum) [reply node-commons/src/main/java/org/apache/iotdb/commons/client/ClientManager.java:61]: Yes

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java [hunk 2]
```diff
@@ -303,14 +308,19 @@ public void MaxWaitClientTimeoutClientManagersTest() throws Exception {
...
-    SyncDataNodeInternalServiceClient syncClient2 = syncClusterManager.borrowClient(endPoint);
-    long end = System.nanoTime();
-    Assert.assertTrue(end - start >= waitClientTimeoutMS * 1_000_000);
-    Assert.assertNull(syncClient2);
+    SyncDataNodeInternalServiceClient syncClient2 = null;
+    try {
+      long start = System.nanoTime();
+      syncClient2 = syncClusterManager.borrowClient(endPoint);
+      long end = System.nanoTime();
+      Assert.assertTrue(end - start >= waitClientTimeoutMS * 1_000_000);
```
[the remark this answers is not in the document]
Comment (zjx1230) [reply node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java:316]: Yes, it is implemented below.
`
public T borrowObject(final K key) throws Exception {
        return borrowObject(key, getMaxWaitDuration().toMillis());
}
`

File: node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java [hunk 3]
```diff
@@ -236,14 +236,19 @@ public void MaxTotalClientManagersTest() throws Exception {
...
-    SyncDataNodeInternalServiceClient syncClient2 = syncClusterManager.borrowClient(endPoint);
-    long end = System.nanoTime();
-    Assert.assertTrue(end - start >= DefaultProperty.WAIT_CLIENT_TIMEOUT_MS * 1_000_000);
-    Assert.assertNull(syncClient2);
+    SyncDataNodeInternalServiceClient syncClient2 = null;
+    try {
+      long start = System.nanoTime();
+      syncClient2 = syncClusterManager.borrowClient(endPoint);
+      long end = System.nanoTime();
+      Assert.assertTrue(end - start >= DefaultProperty.WAIT_CLIENT_TIMEOUT_MS * 1_000_000);
```
[the remark this answers is not in the document]
Comment (zjx1230) [reply node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java:244]: same

File: node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (OneSizeFitsQuorum) [reply node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java:316]: Just as the comment(`get another sync client, should wait waitClientTimeoutMS ms, throw error and return null`) says, the purpose of this test is to test whether the maximum wait time meets expectations. The previous behavior of timeout for `borrowObject` was to return NULL. Now the behavior of timeout for `borrowObject` is to throw an exception. However, today's test logic can no longer test whether the maximum wait time meets expectations, and we need to update the comments too.

[the remark this answers is not in the document]
Comment (zjx1230) [reply node-commons/src/test/java/org/apache/iotdb/commons/ClientManagerTest.java]: Ok, I update the Test.

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

File: node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class SyncThriftClientErrorHandler implements MethodInterceptor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SyncThriftClientErrorHandler.class);
+
+  public static <V extends SyncThriftClient> V newErrorHandlerClient(
```
Comment (qiaojialin) [inline node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java:36]: an ErrorHandler constructs a client is not suitable... 
You could rename this class to SyncThriftClientWithErrorHandler or so

File: node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      Class<V> targetClass, Constructor<V> constructor, Object... args) {
+    Enhancer enhancer = new Enhancer();
+    enhancer.setSuperclass(targetClass);
+    enhancer.setCallback(new SyncThriftClientErrorHandler());
+    if (constructor == null) return (V) enhancer.create();
```
Comment (qiaojialin) [inline node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java:41]:
```suggestion
    if (constructor == null) {
        return (V) enhancer.create();
    }
```

Comment (zjx1230) [reply node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java]: ok

File: node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java (hunk 4 above)
Comment (zjx1230) [reply node-commons/src/main/java/org/apache/iotdb/commons/client/sync/SyncThriftClientErrorHandler.java]: ok

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**